### PR TITLE
Refactor how images are pushed

### PR DESCRIPTION
**Pull request by jpkrohling**
Changed how images are pushed to registries, both for master and releases.

### CODE REVIEW

**Comment by jpkrohling** [comment]
@kevinearls, would you review this one as well?

File: .github/workflows/publish-images.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+      - name: Set env vars for the job
+        run: |
+          grep -v '\#' versions.txt | grep opentelemetry-collector | awk -F= '{print "OTELCOL_VERSION="$2}' >> $GITHUB_ENV
+          echo "VERSION_DATE=$(date -u +'%Y-%m-%dT%H:%M:%SZ')" >> $GITHUB_ENV
+          echo "VERSION=$(git describe --tags | sed 's/^v//')" >> $GITHUB_ENV
+
+      - name: Docker meta
+        id: docker_meta
+        uses: crazy-max/ghaction-docker-meta@v1
```
Comment (kevinearls) [inline .github/workflows/publish-images.yaml:24]: crazy-max?  Should we be depending on someone named crazy-max?  :-)

**Review by kevinearls — APPROVED** [review]
LGTM

File: .github/workflows/publish-images.yaml (hunk 1 above)
Comment (jpkrohling) [reply .github/workflows/publish-images.yaml:24]: I had the exact thought. Unfortunately, the official docs for the Docker action recommends this one dynamic labels and tags.

**Comment by jpkrohling** [comment]
I'm merging, as there's only one way of knowing it works.

**Pull request merged**